WordPress/Classes/Stores/StatsWidgetsStore.swift

```diff
@@ -1,3 +1,4 @@
+import JetpackStatsWidgetsCore
```

Comment:
One could argue that it's not appropriate to import JetpackStatsWidgetsCore in the apps target. After all, the purpose of that target it to hold the business logic _for the widgets_.

The reason this is required is that this code references `HomeWidgetData`.

One way around it would be to move `HomeWidgetData` in a dedicated "shared layer" or "model" Package, that both apps and JetpackStatsWidgetsCore can access. At this time, it seems unnecessary—but happy to change my mind on it.

Again, given the context of this PR, I didn't look into the underlying architecture and why we need to access widgets type at this level in the first place. Maybe there's a better fix waiting to be discovered...

---

The irony is not lost on me that all this work to remove a file from WordPress for the purpose of testing resulted in a whole framework being linked into it. But consider that the way it worked before was to have `HomeWidgetData` as part of the WordPress target. So, this is still an improvement.

<img width="1356" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/ce44bef5-3331-46c1-83ec-d89b80f9b843">

The screenshot shows the file being part of the intents target as well. The fact that the app compiles without us linking the package to the intents target makes me confident that target membership had already become unnecessary.